### PR TITLE
Added `other_uk_qualification_type` to Vendor API

### PR DESCRIPTION
**Pull request by dcyoung-dev**
## Context

This attribute was only shown for GCSEs however the data is relevant to all Qualifications

## Changes proposed in this pull request

Added `other_uk_qualification_type` to Vendor API

## Link to Trello card

https://trello.com/c/1rAVdEiq

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by CatalinVoineag — APPROVED** [review]

File: app/presenters/concerns/qualification_api_data.rb [hunk 1]
```diff
@@ -151,11 +151,17 @@ def structured_gcse_to_hashes(gcse)
 private
 
   def completing_qualification(qualification)
-    return {} unless include_completing_qualification? && qualification.gcse?
+    return {} unless include_completing_qualification?
+
+    if qualification.gcse?
+      return {
+        currently_completing_qualification: qualification[:currently_completing_qualification],
+        missing_explanation: qualification[:missing_explanation],
```
Comment (elceebee) [inline app/presenters/concerns/qualification_api_data.rb:159]: It looks like we were including these fields for non-gcse qualifications in the past. are they always nil?

Comment (dcyoung-dev) [reply app/presenters/concerns/qualification_api_data.rb:159]: Have I messed up my if's and unless' 🤦‍♂️

Comment (dcyoung-dev) [reply app/presenters/concerns/qualification_api_data.rb:159]: We were only including these fields if the qualification was a gcse. We are now including the `other_uk_qualification_type` for all qualifications. 
I've waled through how I refactored this and came to the same code :😅

**Comment by dcyoung-dev** [comment]
I'm having doubts over whether or not this should be part of a v1.5 API 🤔

**Comment by inulty-dfe** [comment]
I find this hard to parse.


```
def completing_qualifications(qualification)
  values = {}
  return values if !include_completing_qualification?

  values = values.merge({other_uk_qualification_type: qualification[:other_uk_qualification_type]})

  if qualification.gcse?
    values = values.merge({
        currently_completing_qualification: qualification[:currently_completing_qualification],
        missing_explanation: qualification[:missing_explanation],
        other_uk_qualification_type: qualification[:other_uk_qualification_type],
      })
  end

 values
end
```

I'm not sure this is better but it feels like there is a cumulation happening on the return value ({}). The duplication of `other_uk_qualification_type` on lines 160 and 165 seems strange.

If you don't take the suggestion above verbatim I still think this could be expressed better.

**Review by inulty-dfe — CHANGES_REQUESTED** [review]
I've added a comment with a change suggestion, I feel this could do with a little more work to make it easier to read

**Comment by dcyoung-dev** [comment]
> I find this change hard to parse. Here is an alternative:
> 
> ```ruby
> def completing_qualifications(qualification)
>   values = {}
>   return values if !include_completing_qualification?
> 
>   values = values.merge({other_uk_qualification_type: qualification[:other_uk_qualification_type]})
> 
>   if qualification.gcse?
>     values = values.merge({
>         currently_completing_qualification: qualification[:currently_completing_qualification],
>         missing_explanation: qualification[:missing_explanation],
>       })
>   end
> 
>  values
> end
> ```
> 
> I'm not sure this is better but it feels like there is a cumulation happening on the return value ({}). The duplication of `other_uk_qualification_type` on lines 160 and 165 seems strange.
> 
> If you don't take the suggestion above verbatim I still think this could be expressed better.

Thanks @inulty-dfe for the feedback - I've went with a slightly different approach, separating out the 2 return paths.👌 Let me know what you think ec64302aa85fd39328fa3bb1984e4d34112b43d9

**Review by inulty-dfe — APPROVED** [review]
Nice refactor, lgtm

File: spec/presenters/vendor_api/v1.4/application_presenter_spec.rb [hunk 2]
```diff
@@ -8,10 +8,28 @@
...
-    subject(:gcse) { attributes[:qualifications][:gcses].first }
+  describe 'qualifications' do
+    context 'when the application has a other_uk_qualification_type qualification' do
+      subject(:other_qualification) { attributes[:qualifications][:other_qualifications].first }
+
+      before do
+        create(
+          :other_qualification,
+          other_uk_qualification_type: 'Equivalency test',
+          application_form: application_form,
```
Comment (avinhurry) [inline spec/presenters/vendor_api/v1.4/application_presenter_spec.rb:19]:
```suggestion
          application_form:,
```

**Review by avinhurry — APPROVED** [review]
